docs/src/device.rst

```diff
@@ -0,0 +1,77 @@
+
+.. _device:
+
+:c:type:`uv_device_t` --- Device handle
+=======================================
+
+Device handles are used to read and write on char/block device.
+
+:c:type:`uv_device_t` is a 'subclass' of :c:type:`uv_stream_t`.
+
+
+Data types
+----------
+
+.. c:type:: uv_device_t
+
+    Device handle type.
+
+.. c:type:: uv_ioargs_t
+
+    Device abstract arg struct, it is related with OS.
+
+    ::
+
+        /* on windows */
+        typedef struct uv_ioargs_s {
+          void* input;
+          uint32_t input_len;
+          void* output;
+          uint32_t output_len;
+        } uv_ioargs_t; 
+
+        /* on ?nux */
+        typedef struct uv_ioargs_s {
+          void* arg;
+        } uv_ioargs_t;
+
+    On windows, `uv_ioargs_t` contains paramaters pass to `DeviceIoControl` 
+    function, please see: https://msdn.microsoft.com/en-us/library/windows/desktop/aa363216(v=vs.85).aspx
+
+    On linux `uv_ioargs_t` conatins arg (an untyped pointer to memory) pass to 
+    `ioctl` at 3rd paramater,  which decided by request type.
+
+
+Public members
+^^^^^^^^^^^^^^
+
+N/A
+
+.. seealso:: The :c:type:`uv_stream_t` members also apply.
+
+
+API
+---
+
+.. c:function::  int uv_device_init(uv_loop_t* handle, uv_device_t* device, const char* path, int flags);
+
+    Initialize a new device with the given path. `path` must point a dev path, 
+    just ``/dev/device_name`` on linux, bsd or osx, and ``\.\Global\DeviceName``
+    or \\.\DeviceName on windows. `flags` must be O_RDONLY,O_WRONLY or O_RDWR.
```

Comment:
On Unix we might need more flags for open, like O_NOCTTY when opening a serial device.